CFLAGS and LDFLAGS were modified for macOS in 2004.
See https://github.com/mono/libgdiplus/commit/bce17c28c445c49e9349aace2eb0e4a9bd235aad.
It is not entirely clear why the changes were needed because the bug report is no longer available.
However, they are causing problems.
See https://trac.macports.org/ticket/24501.

There is an upstream pull request to fix the issue.
See https://github.com/mono/libgdiplus/pull/534.

no_x was removed upstream.
See https://github.com/mono/libgdiplus/commit/bf232eb9d5031d229933c0270cf74f3bdd1ccbc1.

no_libpng16 it not appropriate for MacPorts, which provides libpng16.

--- configure.orig	2018-03-28 12:28:08.000000000 -0700
+++ configure	2019-02-20 15:39:46.000000000 -0700
@@ -13281,10 +13281,6 @@
 #define OSX 1
 _ACEOF
 
-		CFLAGS="-no-cpp-precomp $CFLAGS"
-		LDFLAGS="-flat_namespace -undefined suppress $LDFLAGS"
-		no_x=yes
-		no_libpng16=yes # workaround until we can upgrade the libpng used by bockbuild
 		;;
 
 	*)
@@ -14132,12 +14128,6 @@
 fi
 
 
-# Add xrender here so that we don't fail to find glib if we won't have xrender.pc
-if test "x$no_x" != "xyes" && "x$x11_available" = "xyes"; then
-   GDIPLUS_LIBS="$GDIPLUS_LIBS `$PKG_CONFIG --libs xrender `"
-   GDIPLUS_CFLAGS="$GDIPLUS_CFLAGS `$PKG_CONFIG --cflags xrender `"
-fi
-
 if test x$with_libexif != xno && test -z "$LIBEXIF"; then
 
 pkg_failed=no
